scripts/confirm_sns_subscription.py

```diff
@@ -0,0 +1,101 @@
+"""
+This script prompts the user for an SNS subscription confirmation link and then
+confirms the subscription with AuthenticateOnUnsubscribe enabled. Without
+AuthenticateOnUnsubscribe, anyone with an unsubscribe link would be able to
+delete the subscription. This would be especially undesired in the case of a
+subscription that sends notifications to a group email address.
+"""
+import json
+import logging
+import sys
+from urllib.parse import (
+    parse_qs,
+    urlparse,
+)
+
+from more_itertools import (
+    one,
+)
+
+from azul import (
+    config,
+)
+from azul.deployment import (
+    aws,
+)
+from azul.logging import (
+    configure_script_logging,
+)
+
+log = logging.getLogger(__name__)
+
+sns_client = aws.sns
+
+
+def main():
+    if config.disable_monitoring:
+        print('SNS subscription confirmation skipped, monitoring is disabled.')
+        return 0
+    elif subscription_pending_confirmation():
+        return confirm_subscription()
+    else:
+        print('SNS subscription confirmation skipped, no pending subscriptions found.')
+        return 0
+
+
+def subscription_pending_confirmation():
+    """
+    Return true if there is at least one pending subscription
+    """
+    paginator = sns_client.get_paginator('list_subscriptions')
+    for page in paginator.paginate():
+        for subscription in page['Subscriptions']:
+            if subscription['SubscriptionArn'] == 'PendingConfirmation':
+                return True
+    return False
+
+
+def confirm_subscription():
```

Comment:
Type hints here and elsewhere. 

This should return bool so as to contain the "0 means success" oddity in the main function.

In general, I'm not a big fan of threading success/failure status through return values. It is much easier and pythonic to raise an exception on failure and return None on success. With return values it is so easy to forget to forward the status in a function. For prettier output you can raise a specific type of exception, catch it in the main function, print its argument, and call sys.exit(1). That way you won't get a noisy backtrace in expected failure situations.

Reply:
>  In general, I'm not a big fan of threading success/failure status through return values. It is much easier and pythonic to raise an exception on failure and return None on success.

Refactored the script to return `None` on success or raise an exception, which produces the desired exit code (0 or 1 respectively) to the shell.